cxx_supportlib build failure with clang++-19 due to invalid member access

## Issue report

**Question 1: What is the problem?**

The passenger agent fails to build with clang 19.

Steps to reproduce:

0. Start a fresh Fedora 41 container.
```bash
docker run --workdir /root --rm -it fedora:41 bash
```

1. Install dependencies:
```bash
yes | dnf install git ruby-devel clang++ libcurl-devel
```

2. Download and build Passenger with clang++-19:
```bash
export CC=clang CXX=clang++
git clone --recurse-submodules https://github.com/phusion/passenger
cd passenger
gem build
gem install ./passenger*.gem
passenger-config compile-agent --auto
```

3. The build will fail. The log file is long, so I won't paste it all here, but this is the relevant error:
```
clang++ -o /tmp/passenger-install.4e2qu7/support-binaries/WatchdogMain.o  -Isrc/agent -Isrc/cxx_supportlib -Isrc/cxx_supportlib/vendor-copy -Isrc/cxx_supportlib/vendor-modified -Isrc/cxx_supportlib/vendor-modified/libev -Wno-ambiguous-member-template -DUSE_VENDORED_LIBEV -Isrc/cxx_supportlib/vendor-copy/libuv/include -Isrc/cxx_supportlib/vendor-modified/websocketpp -D_REENTRANT -I/usr/local/include -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wpointer-arith -Wwrite-strings -Wno-long-long -Wno-missing-field-initializers -Wno-unknown-pragmas -Wno-ambiguous-member-template -fvisibility=hidden -DVISIBILITY_ATTRIBUTE_SUPPORTED -DHAS_ALLOCA_H -DHAVE_ACCEPT4 -DHAS_SFENCE -DHAS_LFENCE -DPASSENGER_DEBUG -DBOOST_DISABLE_ASSERTS -g -fno-limit-debug-info -std=gnu++11 -Wno-unused-local-typedefs -Wno-format-nonliteral -DHAS_UNORDERED_MAP -c src/agent/Watchdog/WatchdogMain.cpp
In file included from src/agent/Watchdog/WatchdogMain.cpp:66:
In file included from src/agent/Shared/ApiServerUtils.h:71:
In file included from src/cxx_supportlib/ServerKit/Server.h:61:
src/cxx_supportlib/Algorithms/MovingAverage.h:193:44: error: no member named 'sumOfWeights' in 'DiscExpMovingAverageWithStddev<alpha, alphaTimeUnit, maxAge>'
  193 |                 return sqrt(copy.sumOfSquaredData / copy.sumOfWeights - pow(copy.average(), 2));
      |                                                     ~~~~ ^
1 error generated.
```

**Question 2: Passenger version and integration mode**:

Passenger built from the `stable-6.0` branch, commit c7599c49ef2d757c877045564f26156bb36dc4e5.

**Question 3: OS or Linux distro, platform (including version)**:

Host: Linux 6.1.0-25-amd64, Debian 12
Container: 6.1.0-25-amd64, Fedora 41

**Question 4: Passenger installation method**:

[X] source tarball

**Question 5: Your app's programming language (including any version managers) and framework (including versions)**:
- For example: Ruby 2.3.0, RVM, Rails 5.0.0; Node.js 4.0.0 with Express 4.13.4

N/A

**Question 6: Are you using a PaaS and/or containerization? If so which one?**
- For example: Heroku, Amazon Container Services, Docker 1.9 with an image based on passenger-docker

I am using Docker, but only for convenient access to clang 19.

**Question 7: Anything else about your setup that we should know?**

Nope.

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,6 +1,7 @@
 Release 6.0.25 (Not yet released)
 -------------
  * [Standalone] Changes Passenger (not app) start and stop timeouts to 25s (from 15s) except for Nginx engine mode, which retains a stop timeout of 60s.
+ * Fixes compilation with clang 19 (latest Fedora update) by dropping a buggy stddev function from the moving average header. Closes GH-2580.
  * 
 
 

--- a/src/cxx_supportlib/Algorithms/MovingAverage.h
+++ b/src/cxx_supportlib/Algorithms/MovingAverage.h
@@ -186,12 +186,6 @@ public:
 	double stddev() const {
 		return sqrt(sumOfSquaredData / dema.sumOfWeights - pow(average(), 2));
 	}
-
-	double stddev(unsigned long long now) const {
-		DiscExpMovingAverageWithStddev<alpha, alphaTimeUnit, maxAge> copy(*this);
-		copy.update(0, now);
-		return sqrt(copy.sumOfSquaredData / copy.sumOfWeights - pow(copy.average(), 2));
-	}
 };
 
 

